rs/ai/_example/handlers/event_handler.py

```diff
@@ -1,285 +1,20 @@
-                return "choose refuse"
-            if state.deck.contains_card_amount("strike") >= 3:  # note: these are specifically un-upgraded strikes
-                if state.has_relic("Blood Vial"):
-                    return "choose 1"  # Nom the Spire
-                else:
-                    return "choose accept"
-            return "choose refuse"
-
-        # Act 2, 3
-
-        if event_name == "Designer In-Spire":
-            return "choose 0"  # It'll do reasonable things like upgrading and removing.
-
-        # Act 3
-
-        if event_name == "Falling":
-            options = state.get_falling_event_options()
-
-            # check for stuff we want to purge
-            for least_desired in CARD_REMOVAL_PRIORITY_LIST:
-                if least_desired in options:
-                    for idx, card in enumerate(options):
-                        if card == least_desired:
-                            return "choose " + str(idx)
-
-            # check for cards not in the pickup list
-            for idx, option in enumerate(options):
-                if option not in DESIRED_CARDS_FOR_DECK:
-                    return "choose " + str(idx)
-
-            # check for lowest card on the pickup list
-            pickup_prios = list(DESIRED_CARDS_FOR_DECK.keys())
-            pickup_prios.reverse()
-
-            for least_desired in pickup_prios:
-                if least_desired in options:
-                    for idx, card in enumerate(options):
-                        if card == least_desired:
-                            return "choose " + str(idx)
-
-        if event_name == "Mind Bloom":
-            return "choose 0"  # Fight an Act 1 boss for a relic.
-
-        if event_name == "The Moai Head":
-            return "choose 1"  # Get a bunch of money or leave
-
-        if event_name == "Mysterious Sphere":
-            if hp_per >= 70:
-                return "choose 0"  # Get 'em!
-            else:
-                return "choose 1"  # Leave
-
-        if event_name == "Secret Portal":
-            return "choose 1"  # Nope - we want the rest of Act 3 to not screw up our stats.
-
-        if event_name == "Sensory Stone":
-            return "choose 0"  # Get a free colorless obtain
-
-        if event_name == "Tomb of Lord Red Mask":
-            if state.has_relic("Red Mask") or state.game_state()['gold'] <= 130:
-                return "choose 0"  # Either free money or cheap.
-            else:
-                return "choose 1"  # Leave
+        event = state.get_event()
 
-        if event_name == "Winding Halls":
-            if state.get_relic_counter("Omamori") >= 1 and hp_per < 75:
-                return "choose 1"  # Take the curse and heal
-            if hp_per <= 10:
-                return "choose 1"  # Take the curse and heal
-            return "choose 2"  # Lose Max HP
+        # Custom event handling specifically for this strategy
+        match event:
```

Comment:
I... did not know python had this 😄 . Cool.

Reply:
It seems to be semi-new, in that it's been around for a few years but people are still discovering it 😄